### PR TITLE
time-util,format-table: add relative_monotonic variant for timestamp

### PR DESCRIPTION
**Pull request by YHNdnzj**
Follow-up for #27769

Addresses https://github.com/systemd/systemd/pull/27769#discussion_r1205215743

### CODE REVIEW

File: src/basic/time-util.c [hunk 1]
```diff
@@ -422,22 +422,28 @@ char *format_timestamp_style(
...
+        assert(!monotonic || !implicit_left);
 
         if (!timestamp_is_set(t))
                 return NULL;
 
-        n = now(CLOCK_REALTIME);
-        if (n > t) {
-                d = n - t;
+        if (monotonic) {
+                d = t;
```
Comment (poettering) [inline src/basic/time-util.c]: hmm? why not subtract this from now(CLOCK_MONOTONIC)? i am confused now.

Comment (YHNdnzj) [reply src/basic/time-util.c]: (Why did I write something like that half a day ago...)

File: src/basic/time-util.c [hunk 2]
```diff
@@ -422,7 +422,7 @@ char *format_timestamp_style(
         return buf;
 }
 
-char* format_timestamp_relative_full(char *buf, size_t l, usec_t t, bool implicit_left) {
+char* format_timestamp_relative_full(char *buf, size_t l, usec_t t, bool implicit_left, bool monotonic) {
```
Comment (yuwata) [inline src/basic/time-util.c]: Not a matter, but I's like to make this take CLOCK_XYZ instead of boolean flag.

**Comment by YHNdnzj** [comment]
@mrc0mmand CentOS CI (Arch Linux) seems to be stuck in `TEST-01-BASIC`? Also in #27746 (https://jenkins-systemd.apps.ocp.cloud.ci.centos.org/job/upstream-vagrant-archlinux/2660/). Any ideas?

**Comment by mrc0mmand** [comment]
> @mrc0mmand CentOS CI (Arch Linux) seems to be stuck in `TEST-01-BASIC`? Also observed in #27746 (https://jenkins-systemd.apps.ocp.cloud.ci.centos.org/job/upstream-vagrant-archlinux/2660/). Any ideas?

This is happening across multiple recent PRs and I guess it's the PID1 deadlock issue (can't link it from mobile) that got exacerbated by the recently introduced test for utmp stuff that does a lot of daemon-reexecs.

**Comment by poettering** [comment]
excellent! thanks!

**Review by bluca — APPROVED** [review]

**Comment by yuwata** [comment]
> > @mrc0mmand CentOS CI (Arch Linux) seems to be stuck in `TEST-01-BASIC`? Also observed in #27746 (https://jenkins-systemd.apps.ocp.cloud.ci.centos.org/job/upstream-vagrant-archlinux/2660/). Any ideas?
> 
> This is happening across multiple recent PRs and I guess it's the PID1 deadlock issue (can't link it from mobile) that got exacerbated by the recently introduced test for utmp stuff that does a lot of daemon-reexecs.

Filed as #27807.

**Pull request merged**